### PR TITLE
Update Auth0 API requirements

### PR DESCRIPTION
**Pull request by fabien0102**
ref #6 

Hello everybody,
the Auth0 API seams to have changed since this example (no more `id_token` in the basic response), so this is my fixes after some researches and tests.

Please note that I've fixed the context of the handler, I can make another PR for this if needed 😉

Auth0 doc ref:
https://auth0.com/docs/api-auth/intro#implicit-grant

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -28,13 +28,14 @@ class App extends React.Component {
...
-    Linking.addEventListener('url', this._handleAuth0Redirect);
+    Linking.addEventListener('url', this._handleAuth0Redirect.bind(this));
   }
 
   _loginWithAuth0 = async () => {
     const redirectionURL = `${auth0Domain}/authorize` + this._toQueryString({
       client_id: auth0ClientId,
-      response_type: 'token',
+      response_type: 'id_token',
+      nonce: 'alongrandomstringtopreventtokenreplayattacks',
```
Comment (ice-chillios) [inline main.js]: Maybe this should be handled by exactly random string? :) [Auth0 example](https://auth0.com/docs/api-auth/tutorials/nonce) seems to be handling this in nicer way :)

**Review by ice-chillios — COMMENTED** [review]
@fabien0102 Thanks for updating :) I've gone into native solution at the end because nobody was responding...

File: main.js (hunk 1 above)
Comment (fabien0102) [reply main.js]: I wanted to keep this PR simple, but yeah a random string store into asyncStorage is better for production requirements ;) So as you want, I can add this security part or not

**Comment by fabien0102** [comment]
Normally now it's working and it's really secured!

I let @charlesvinette make the real integration test (it's his account on this example)

File: main.js [hunk 2]
```diff
@@ -27,25 +28,28 @@ class App extends React.Component {
...
     username: undefined,
   };
+
   componentDidMount() {
-    Linking.addEventListener('url', this._handleAuth0Redirect);
+    Linking.addEventListener('url', this._handleAuth0Redirect.bind(this));
   }
 
-  _loginWithAuth0 = async () => {
+  async _loginWithAuth0() {
```
Comment (ice-chillios) [inline main.js]: Why You've changed arrow functions into normal ones? When you are changing this from arrow you do not have a `this` context. That's why you have to `.bind(this)` at the top in `this._handleAuth0Redirect` call :)

Comment (fabien0102) [reply main.js]: Good to know, I just would to uniformise all functions style ^^ So back to arrow functions!

**Comment by charlesvinette** [comment]
Hello guys! 

Thanks for helping out with the example. With the recent release of Expo SDK 21, we released a new module called AuthSession (https://docs.expo.io/versions/latest/sdk/auth-session.html) to facilitate Auth flow with Webbrowsers, such as Auth0. From now on, I would suggest using this approach, as seen in this PR that I am about to review -> https://github.com/expo/auth0-example/pull/10. Once again, thanks for your involvement and let me know what you think of the new module!

**Review by charlesvinette — COMMENTED** [review]
Hello guys!

Thanks for helping out with the example. With the recent release of Expo SDK 21, we released a new module called AuthSession (https://docs.expo.io/versions/latest/sdk/auth-session.html) to facilitate Auth flow with Webbrowsers, such as Auth0. From now on, I would suggest using this approach, as seen in this PR that I am about to review -> #10. Once again, thanks for your involvement and let me know what you think of the new module!

**Comment by fabien0102** [comment]
Nice! I have just finished to test this pattern into my own playground app ^^ I will try this new expo module tomorrow! BTW, I close this PR 😉

**Comment by charlesvinette** [comment]
@fabien0102 Glad to hear it! Have a good day and let me know if I can be of any help

**Comment by fabien0102** [comment]
@charlesvinette 
Results of the testing with the expo `AuthSession`:

- We need to add the `nonce` to be compliante with the last security requirements. 
- And I have this warning screen before auth0 page… How can I skip it?

image
image

**Comment by charlesvinette** [comment]
Hey @fabien0102 ! Can you share your code where you initiate the `AuthSession`? And for your second question, this prompt will present itself when the redirect URL does not match the published or standalone slug of your application, you can read more about why here -> https://docs.expo.io/versions/latest/sdk/auth-session.html#it-makes-redirect-url-whitelists-easier-to-manage-for-development-and-working-in-teams

**Comment by fabien0102** [comment]
@charlesvinette Thanks for this point. I made a PR ( #11 ) to add the missing nonce (and my local is exactly the same as this example for this part, it's not yet public but soon 😉 ).

I think it can works with turning on the OIDC part without nonce (but I'm really not sure and I prefer not play with this button ^^)

image

**Pull request closed without merging**